sdk/wallet-adapter/adapters/wallet-standard-adapter/src/StandardWalletAdapter.ts

```diff
@@ -89,23 +102,44 @@ export class StandardWalletAdapter implements WalletAdapter {
     }
   }
 
-  async signTransaction(transaction: SignableTransaction) {
+  signTransaction: WalletAdapter["signTransaction"] = (transactionInput) => {
+    let chain = transactionInput.chain || transactionInput.account.chains[0];
+    if (!chain) {
+      throw new Error("Transaction chain not found");
+    }
+    const version = this.#wallet.features["sui:signTransaction"].version;
+    if (!isFeatureCompatible(version, suiSignTransactionLatestVersion)) {
+      throw new Error(
+        `Version mismatch, signTransaction feature version ${version} is not compatible with version ${suiSignTransactionLatestVersion}`
+      );
+    }
     return this.#wallet.features["sui:signTransaction"].signTransaction({
-      transaction,
+      ...transactionInput,
+      chain,
     });
-  }
+  };
 
-  async signAndExecuteTransaction(
-    transaction: SignableTransaction,
-    options?: { requestType?: ExecuteTransactionRequestType }
-  ) {
+  signAndExecuteTransaction: WalletAdapter["signAndExecuteTransaction"] = (
+    transactionInput
+  ) => {
+    let chain = transactionInput.chain || transactionInput.account.chains[0];
```

Comment:
Instead of having this defaulting on the adapter level, we could shuffle it to the wallet kit core level, which might make more sense given we'll also have things like account defaults there.